schema/token.graphql

```diff
@@ -0,0 +1,409 @@
+
+type SaleTransaction @entity @index(fields: ["sale", "account"]) {
+  "counter"
+  id: ID!
+
+  "amount of token bought"
+  quantity: BigInt!
+
+  "sale"
+  sale: Sale!
+
+  "buyer account"
+  account: TokenAccount!
+
+  "block"
+  createdIn: Int!
+}
+
+type AmmCurve @entity {
+  "counter"
+  id: ID!
+
+  "token this Amm is for"
+  token: Token!
+
+  "quantity bought on the market by the amm"
+  burnedByAmm: BigInt!
+
+  "quantity sold to the market"
+  mintedByAmm: BigInt!
+
+  "the amm slope parameter a in the formula a * x + b"
+  ammSlopeParameter: BigInt!
+
+  "the amm intercept parameter b in the formula a * x + b"
+  ammInitPrice: BigInt!
+
+  "transaction for this amm"
+  transactions: [AmmTransaction!] @derivedFrom(field: "amm")
+
+  "finalized (i.e. closed)"
+  finalized: Boolean!
+}
+
+type AmmTransaction @entity @index(fields: ["account", "amm"]) {
+  "counter"
+  id: ID!
+
+  "amount of token bought/sold"
+  quantity: BigInt!
+
+  "total HAPI paid/received for the quantity"
+  pricePaid: BigInt!
+
+  "Reference to the Amm Sale"
+  amm: AmmCurve!
+
+  "buyer account"
+  account: TokenAccount!
+
+  "price per unit in HAPI"
+  pricePerUnit: BigInt!
+
+  "was it bought/sold"
+  transactionType: AmmTransactionType!
+
+  "block"
+  createdIn: Int!
+}
+
+enum AmmTransactionType {
+  BUY
+  SELL
+}
+
+type TokenAccount @entity @index(fields: ["token", "member"]) {
+  "counter"
+  id: ID!
+
+  "member id ref"
+  member: Membership!
+
+  "token for which the account is created"
+  token: Token!
+
+  "staked amount, to avoid walking to split participations"
+  stakedAmount: BigInt!
+
+  "total token amount (not necessarely liqud) for the account"
+  totalAmount: BigInt!
+
+  "sale transactions for this account"
+  saleTransactions: [SaleTransaction!] @derivedFrom(field: "account")
+
+  "amm transactions for this account"
+  ammTransactions: [AmmTransaction!] @derivedFrom(field: "account")
+
+  "information about token account's participation in revenue shares"
+  revenueShareParticipation: [RevenueShareParticipation!] @derivedFrom(field: "account")
+
+  "vesting schedule info, it can be empty if no vesting schedules are pending"
+  vestingSchedules: [VestedAccount!]! @derivedFrom(field: "account")
+
+  "has been deleted or not"
+  deleted: Boolean!
+}
+
+type VestedAccount @entity {
+  "counter"
+  id: ID!
+
+  "vesting schedule reference"
+  vesting: VestingSchedule!
+
+  "account reference"
+  account: TokenAccount!
+
+  "total amount the schedule is vested"
+  totalVestingAmount: BigInt!
+
+  "vesting source"
+  vestingSource: VestingSource!
+}
+
+type SaleVestingSource @variant {
+  sale: Sale!
+}
+type IssuerTransferVestingSource @variant {
+  phantom: Int
+}
+type InitialIssuanceVestingSource @variant {
+  phantom: Int
+}
+
+union VestingSource = InitialIssuanceVestingSource | IssuerTransferVestingSource | SaleVestingSource
+
+type VestingSchedule
+  @entity
+  @index(fields: ["cliffBlock", "vestingDurationBlocks", "cliffPercent"]) {
```

Comment:
Due to a bug in Subsquid `typeorm-codegen` this index is causing some issues with the codegen. I'd say let's just drop it.